Assets/VRTK/Scripts/Locomotion/VRTK_GazeDrivenLocomotion.cs

```diff
@@ -0,0 +1,104 @@
+    /// </remarks>
+    public class VRTK_GazeDrivenLocomotion: MonoBehaviour
+    {
+        [Tooltip("The speed of the character when HMD gaze is at the edge of the game pad")]
+        [SerializeField]
+        private float maxSpeed = 2.0f;
+        [Tooltip("The directionpad UI that will show the user where their gaze is and how it's affected the locomotion of the character. Use the DirectionPad Prefab in VRTK/Prefabs")]
+        [SerializeField]
+        private GameObject directionPad;
+        [Tooltip("The character game object that you want to locomote.")]
+        [SerializeField]
+        private GameObject characterModel;
+        [Tooltip("Sets the size of the gamepad that spawns underneath the player, which affects the sensitivity and where users look to drive movement of character")]
+        [SerializeField]
+        private float gamePadSize;
+        [Tooltip("The character should be attached to an empty game object, so that certain objects that should not rotate with the character can be attached the empty root object")]
+        [SerializeField]
+        private GameObject characterController;
+        private Quaternion characterLookRotation;
+        private Vector3 directionVector = new Vector3(0, 0, 0);
+        private GameObject interactionPlane;
+        private Slider slider;
+        private Transform hmd; 
+        private void Start()
+        {
+            hmd = VRTK_DeviceFinder.HeadsetCamera();
+            CreateInteractionPlane();
+            CreateDirectionPadUI();
+        }
+        private void Update()
+        {
+            Ray ray = new Ray(hmd.position, hmd.forward);
+            RaycastHit hit;
+            if (Physics.Raycast(ray, out hit))
+            {
+                GameObject collisionObject = hit.transform.gameObject;
+                if (hit.collider.gameObject == interactionPlane)
+                {
+                    Vector3 localizedHitCoordinates = hit.transform.InverseTransformPoint(hit.point);
+                    float x = localizedHitCoordinates.x / ((.5f));
+                    float y = localizedHitCoordinates.y / ((.5f));
+                    directionVector = new Vector3(x, 0, y);
+                    Vector3 directionalUnitVector = (hit.point - directionPad.transform.position).normalized;
+                    UpdateDirectionPadUI(directionalUnitVector, Mathf.Clamp(directionVector.magnitude, 0, 1));
+                    Vector3 CharacterLookVector = (hit.point - characterModel.transform.position).normalized;
+                    CharacterLookVector.y = 0;
+                    characterLookRotation = Quaternion.LookRotation(CharacterLookVector);
+                    characterController.transform.Translate(CalculateSpeedVector(directionVector));
+                }
+                //otherwise stop it
+                else
+                {
+                    directionVector = Vector3.zero;
+                }
+            }
+        }
+        private void LateUpdate()
```

Comment:
All MonoBehaviour methods need to be `protected virtual` so people can extend the scripts if required